internal/mode/static/handler.go

```diff
@@ -511,6 +496,42 @@ func getGatewayAddresses(
 	return gwAddresses, nil
 }
 
+// setDeploymentCtx sets the deployment context metadata for nginx plus reporting.
+func (h *eventHandlerImpl) setDeploymentCtx(ctx context.Context, cfg *dataplane.Configuration) error {
+	if !h.cfg.plus {
+		return nil
+	}
+
+	podNSName := types.NamespacedName{
+		Name:      h.cfg.gatewayPodConfig.Name,
+		Namespace: h.cfg.gatewayPodConfig.Namespace,
+	}
+
+	clusterInfo, err := telemetry.CollectClusterInformation(ctx, h.cfg.k8sReader)
+	if err != nil {
+		return fmt.Errorf("error getting cluster information")
+	}
+
+	replicaSet, err := telemetry.GetPodReplicaSet(ctx, h.cfg.k8sReader, podNSName)
+	if err != nil {
+		return fmt.Errorf("failed to get replica set for pod %v: %w", podNSName, err)
+	}
+
+	deploymentID, err := telemetry.GetDeploymentID(replicaSet)
+	if err != nil {
+		return fmt.Errorf("failed to get NGF deploymentID: %w", err)
+	}
+
+	cfg.DeploymentContext = dataplane.DeploymentContext{
+		Integration:      "ngf",
+		ClusterID:        clusterInfo.ClusterID,
```

Comment:
Probably not, but I don't think I want to have this field just passed around everywhere that it might be needed.

Otherwise, we'd need a new object to store this value in with a Getter. IMO not worth the effort right now.